lupa/_lupa.pyx

```diff
@@ -1908,19 +1908,19 @@ cdef int py_enumerate(lua_State* L) nogil:
     if lua.lua_gettop(L) > 2:
         lua.luaL_argerror(L, 3, "invalid arguments")   # never returns!
     cdef py_object* py_obj = unpack_python_argument_or_jump(L, 1)
-    cdef double start = lua.lua_tonumber(L, -1) if lua.lua_gettop(L) == 2 else 0.0
+    cdef lua.lua_Integer start = lua.lua_tointeger(L, -1) if lua.lua_gettop(L) == 2 else 0
```

Comment:
Lua 5.2+ has `lua_tointegerx()`, which reports success or failure with an additional return argument. It seems worth wrapping that in a helper function to raise appropriate exceptions for overflows etc.
A fallback for 5.1 could use a C macro to map it to `lua_tointeger()` and set the error argument to 1 (i.e. it was a number).

Reply:
Thanks!